### PR TITLE
Catch up with some changes in upstream pub

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
It's possible that we'll need to pick up a fix in upstream pub at dart-lang/pub#3736. This PR gets ready for that by catching up with some recent changes in upstream pub.

### CODE REVIEW

**Comment by jeffwidman** [comment]
This would benefit from a review from @jonasfj or @sigurdm

File: pub/helpers/bin/dependency_services.dart [hunk 1]
```diff
@@ -14,7 +14,7 @@ import 'package:pub/src/log.dart' as log;
 class _DependencyServicesCommandRunner extends CommandRunner<int>
     implements PubTopLevel {
   @override
-  String? get directory => argResults['directory'];
```
Comment (jeffwidman) [inline pub/helpers/bin/dependency_services.dart:17]: why is this needed?

**Review by jeffwidman — APPROVED** [review]

File: pub/helpers/bin/dependency_services.dart (hunk 1 above)
Comment (sigurdm) [reply pub/helpers/bin/dependency_services.dart:17]: The CommandRunner class was updated: https://github.com/dart-lang/pub/pull/3587

**Comment by sigurdm** [comment]
One thing this could use is a test of our hashes in pubspec.lock.

They should get added if they are already there, and left out if not present.

This was coded in the dependency_services here:
https://github.com/dart-lang/pub/pull/3482/files#diff-73780674d4f6139a45323ccb874a95106765df0d47fc0b217aad4b22011f0bc3

**Comment by deivid-rodriguez** [comment]
Nice, thanks for pointing that out.

> They should get added if they are already there, and left out if not present.

This is indeed the ideal behavior for us ❤️.

I'll try to add some specs for that!

**Comment by sigurdm** [comment]
In the light of https://github.com/flutter/flutter/issues/119091 we should do a double check of the behavior around the hosted url.

I'm afraid (but haven't tested) that with this PR dependabot would update the hosted url from 'pub.dartlang.org' to 'pub.dev'

Ideally this transition should be done by pub itself IMO. We probably need special handling in the dependency_services.

**Comment by sigurdm** [comment]
> I'm afraid (but haven't tested) that with this PR dependabot would update the hosted url from 'pub.dartlang.org' to 'pub.dev'

Sorry, I never got back to this.

https://github.com/dart-lang/pub/pull/3846 should fix this issue, then we can forward all the way to that.

**Comment by sigurdm** [comment]
The PR has landed now, we should be able to forward the pub ref to: https://github.com/dart-lang/pub/commit/29a7c0990cf070aa118570554f89402c37b54cf6

**Comment by palisadoes** [comment]
Hello everyone, when could this be merged? We have to manually do all dependabot suggested updates to our packages and fixing the underlying issue would help greatly.

**Comment by deivid-rodriguez** [comment]
Closing in favor of #7265.
